src/integration-test/java/com/commercetools/sync/integration/externalsource/products/ProductSyncIT.java

```diff
@@ -179,6 +184,43 @@ void sync_withNewProduct_shouldCreateProduct() {
         assertThat(warningCallBackMessages).isEmpty();
     }
 
+    @Test
+    void sync_withMissingPriceChannel_shouldCreateProductDistributionPriceChannel() {
+        PriceDraftDsl priceDraftWithMissingChannelRef = PriceDraftBuilder.of(MoneyImpl.of("20", "EUR"))
+                .channel(ResourceIdentifier.ofId("missingId")).build();
+
+        ProductVariantDraftDsl masterVariantDraft = ProductVariantDraftBuilder.of(
+                ProductVariantDraftDsl.of()
+                        .withKey("v2")
+                        .withSku("1065833")
+                        .withPrices(Collections.singletonList(priceDraftWithMissingChannelRef)))
+                .build();
+
+        final ProductDraft productDraft = createProductDraftBuilder(PRODUCT_KEY_2_RESOURCE_PATH,
+                ProductType.referenceOfId(productType.getKey()))
+                .masterVariant(masterVariantDraft)
+                .taxCategory(null)
+                .state(null)
+                .build();
+
+
+        final Consumer<String> warningCallBack = warningMessage -> warningCallBackMessages.add(warningMessage);
+
+        ProductSyncOptions syncOptions = ProductSyncOptionsBuilder.of(CTP_TARGET_CLIENT)
+                .errorCallback(this::collectErrors)
+                .warningCallback(warningCallBack)
+                .ensurePriceChannels(true)
+                .build();
+
+        final ProductSync productSync = new ProductSync(syncOptions);
+        final ProductSyncStatistics syncStatistics = executeBlocking(productSync.sync(singletonList(productDraft)));
+
```

Comment:
It is possible, to check if the product of the target project really has the synced price with the required channelrole.

Reply:
de21697658fc7e5b6fa2828b295d98f7b57dc8c3